### PR TITLE
Feature/add address entry page

### PR DESCRIPTION
**Pull request by ryanpaixao**
New things added:
1) Added stage2-address step
2) Incorporated local validation of text fields
3) Updated some of the NPM dependencies that had security issues

*Note: I added state to the StageForm component to keep track of what is displayed in the fields. I have the store being updated if all the fields pass validation.

### CODE REVIEW

File: src/components/StageForm.js [hunk 1]
```diff
@@ -1,78 +1,98 @@
...
+
+    newState[e.target.name] = e.target.value;
+    this.setState(newState);
   }
 
   render() {
     return (
       <Form>
         {Object.entries(this.props.form).map(([key, value]) => {
+          let errorName = key + "_error";
```
Comment (alokito) [inline src/components/StageForm.js]: This constitutes a small bit of repeated code... might be clearer to introduce a "makeErrorKey" function
`function makeErrorKey(key) {
  return key + "_error";
}`
and use it in setupState(), handleChange() and render() to emphasize they are all related.

**Comment by alokito** [comment]
One thing I noticed: when I run "npm i" I get the following message:

> voteremote.us git:(feature/add-address-entry-page) ✗ npm i    
npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN joi-validation-strategy@0.3.3 requires a peer of joi@>=5.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN babel-loader@6.4.1 requires a peer of webpack@1 || 2 || ^2.1.0-beta || ^2.2.0-rc but none is installed. You must install peer dependencies yourself.

Later, while trying to run the project I get

> ./src/components/StageForm.js
Module not found: Can't resolve 'joi'

Running `npm i --save joi` seems to fix this... so maybe add `joi` to package.json?

Also... The "Previous" buttons don't always preserve information, but perhaps that should be for another pull request.

**Comment by ryanpaixao** [comment]
I've fixed the Joi, and field information preservation issue in the 3 latests commits.

File: src/App.js [hunk 2]
```diff
@@ -23,9 +23,9 @@ class App extends Component {
 
     this.stepProps = { getStore: this.getStore.bind(this), updateStore: this.updateStore.bind(this) }
     this.steps = [
-      { name: "Welcome", component: <StageForm form={welcomeStage} {...this.stepProps} /> },
-      { name: "Your Address", component: <StageForm form={addressStage} {...this.stepProps} /> },
-      { name: "You're Done!", component: <StepFinal {...this.stepProps} /> }
+      { name: "Welcome", component: <StageForm step={0} form={welcomeStage} {...this.stepProps} /> },
+      { name: "Your Address", component: <StageForm step={1} form={addressStage} {...this.stepProps} /> },
+      { name: "You're Done!", component: <StepFinal step={3} {...this.stepProps} /> }
```
Comment (alokito) [inline src/App.js]: Any reason in particular for skipping step 2? Are you planning ahead for the vote.org iframe?

File: src/components/StageForm.js [hunk 3]
```diff
@@ -12,25 +12,38 @@ class StageForm extends Component {
...
-      newState[errorName] = null;
+      newState[key] = store[key] === undefined ? "" : store[key];
     });
 
     return newState;
   }
 
+  componentDidUpdate(prevProps) {
+    // If step has changeds
+    if (prevProps.step !== this.props.step) {
```
Comment (alokito) [inline src/components/StageForm.js]: I believe this is the only place step is used... probably you could get by with just comparing prevProps.form with this.props.form and remove step.

**Comment by alokito** [comment]
So... I think it might be worth removing step and just comparing form, but I don't want to hold things up too much. Now sure how/if I can approve this pull request (I'm used to bitbucket, not GitHub).

Looks like I can add a task list here,
- [ ] Remove step and just compare form.

**Comment by ryanpaixao** [comment]
Hey @alokito, I pushed up a new commit that removed the step props, and the step comparison. It now uses a form comparison.

**Pull request merged**